### ISSUE TITLE
ref_option_ref should not trigger for mutable references

### ISSUE THREAD

**Issue by michalburger1** [rust-lang/rust-clippy#9682]
### Summary

The lint tells you to use `Option<&T>` instead of `&Option<&T>` because the former is `Copy` so there's no point in passing around references. However, the lint seems to also triggers for `&Option<&mut T>` and tells you to use `Option<&mut T>` instead, which is not `Copy`.

### Lint Name

ref_option_ref

### Reproducer

I tried this code:

```rust
fn foo(arg: &Option<&mut String>) {
    println!("{:?}", arg);
}
```

I saw this happen:

```
since `&` implements the `Copy` trait, `&Option<&T>` can be simplified to `Option<&T>`
```

I expected to see this happen:

No lint warnings.

### Version

```text
rustc 1.64.0 (a55dd71d5 2022-09-19)
binary: rustc
commit-hash: a55dd71d5fb0ec5a6a3a9e8c27b2127ba491ce52
commit-date: 2022-09-19
host: x86_64-pc-windows-msvc
release: 1.64.0
LLVM version: 14.0.6
```

### Additional Labels

@rustbot label +l-suggestion-causes-error

**Comment by michalburger1** [comment]
Applying the suggestion would cause an error if you're also trying to use the value in other places (it can't be copied so it will be moved).

**Comment by kraktus** [comment]
[Can't reproduce on nightly](https://play.rust-lang.org/?version=nightly&mode=debug&edition=2021&gist=bca5d070d4f30d5c5974504801144abb) so I assume it's been fixed

**Comment by Alexendoo** [comment]
It's a pedantic lint so you'll need to enable it

**Comment by kraktus** [comment]
right my bad!

@rustbot claim
